scripts/cloudformation/deploy.sh

```diff
@@ -22,67 +177,46 @@ else
-if [[ "${CFN_TEMPLATE_FILENAME==}" == "" ]] || [[ "${CFN_TEMPLATE_FILENAME==}" == "''" ]] || [[ "${CFN_TEMPLATE_FILENAME==}" == "None" ]]; then
-  >&2 echo "{\"error\":\"$CFN_TEMPLATE_FILENAME config is required in bitops config.Exiting...\"}"
-  exit 1
-fi
-
-# Exit if CFN Template Parameters Filename is not found
-if [[ "${CFN_PARAMS_FLAG}" == "True" ]] || [[ "${CFN_PARAMS_FLAG}" == "true" ]]; then
-  if [[ "${CFN_TEMPLATE_PARAMS_FILENAME}" == "" ]] || [[ "${CFN_TEMPLATE_PARAMS_FILENAME}" == "''" ]] || [[ "${CFN_TEMPLATE_PARAMS_FILENAME}" == "None" ]]; then
-    >&2 echo "{\"error\":\"$CFN_TEMPLATE_FILENAME config is required in bitops config.Exiting...\"}"
-    exit 1
-  fi
-fi
-
-echo "cd cloudformation Root: $CLOUDFORMATION_ROOT"
-cd $CLOUDFORMATION_ROOT
-
-# Combine parameters
-if [[ "$CFN_MERGE_PARAMETER" == "true" ]] || [[ "$CFN_MERGE_PARAMETER" == "True" ]]; then
-  echo "Combining json files in $CFN_MERGE_DIRECTORY folder"
-  # All files in the $CFN_MERGE_DIRECTORY will be merged into the $CFN_TEMPLATE_PARAMS_FILENAME, if $CFN_TEMPLATE_PARAMS_FILENAME is unset it will use parameters.json
-  COMBINE_FILES=
-  for filename in $(ls $CLOUDFORMATION_ROOT/$CFN_MERGE_DIRECTORY); do
-    COMBINE_FILES+="$CLOUDFORMATION_ROOT/$CFN_MERGE_DIRECTORY/$filename "
-  done;
-  jq '.[]' $COMBINE_FILES | jq -s . > $CFN_TEMPLATE_PARAMS_FILENAME
-fi
-
-# cloud provider auth
-echo "cloudformation auth cloud provider"
-bash $SCRIPTS_DIR/aws/sts.get-caller-identity.sh
-
-# always run cfn template validation first
-if [[ "${CFN_TEMPLATE_VALIDATION}" == "True" ]] || [[ "${CFN_TEMPLATE_VALIDATION}" == "true" ]]; then
-  echo "Running Cloudformation Template Validation"
-  bash $SCRIPTS_DIR/cloudformation/cloudformation_validate.sh "$CFN_TEMPLATE_FILENAME"
-fi
-
-if [[ "${CFN_STACK_ACTION}" == "deploy" ]] || [[ "${CFN_STACK_ACTION}" == "Deploy" ]]; then
-  echo "Running Cloudformation Deploy Stack"
-  bash $SCRIPTS_DIR/cloudformation/cloudformation_deploy.sh "$CFN_TEMPLATE_FILENAME" "$CFN_PARAMS_FLAG" "$CFN_TEMPLATE_PARAMS_FILENAME" "$CFN_STACK_NAME" "$CFN_CAPABILITY" "$CFN_TEMPLATE_S3_BUCKET" "$CFN_S3_PREFIX"
-fi
-
-if [[ "${CFN_STACK_ACTION}" == "delete" ]] || [[ "${CFN_STACK_ACTION}" == "Delete" ]]; then
-  echo "Running Cloudformation Delete Stack"
-  bash $SCRIPTS_DIR/cloudformation/cloudformation_delete.sh "$CFN_STACK_NAME"
-fi
-
-# Check for After Deploy Scripts
-bash $SCRIPTS_DIR/deploy/after-deploy.sh "$CLOUDFORMATION_ROOT"
\ No newline at end of file
+multi_region_targets="$(bash "$SCRIPTS_DIR/bitops-config/get.sh" "$CLOUDFORMATION_BITOPS_CONFIG" "cloudformation.multi-regional-target-regions" "")"
+
+if [[ -n $multi_region_targets ]]; then
+  # ~ # ~ MULTI REGION ~ # ~ # 
+  echo "Using Multi-Regional deployment strategy"
+
+  # sync the templates folder
+  run_s3_sync_templates
+  
+  for region in $(echo $multi_region_targets);do
+    if [[ $region == "-" ]]; then
+      # This accounts for the regions being in a list 
+      continue
+    
+    else
```

Comment:
This seems like a dirty way of looping through an array.

Check out `install_tools.sh` It reads a yaml array:
```
export TERRAFORM_VERSIONS=$(cat build.config.yaml | shyaml get-values terraform.versions)

    while IFS='' read -r version; do
        TERRAFORM_DOWNLOAD_URL="https://releases.hashicorp.com/terraform/${version}/terraform_${version}_linux_amd64.zip"
        echo ${TERRAFORM_DOWNLOAD_URL}
        curl -LO ${TERRAFORM_DOWNLOAD_URL} && unzip terraform_${version}_linux_amd64.zip -d ./
        mv terraform /usr/local/bin/terraform-${version}
        chmod +x /usr/local/bin/terraform-${version}
    done <<< "$TERRAFORM_VERSIONS"
```
https://github.com/bitovi/bitops/blob/master/scripts/setup/install_tools.sh#L19-L25